File: TopCoder/python/arr/RemoveDuplicates.py

class Solution:
    def removeDuplicates(self, nums) -> int:
        """
        start_index = 1
        target_index = 1
        number_to_search = nums[1]

        while start_index < len(nums):
            if number_to_search == nums[start_index]:
                start_index = start_index + 1
                continue
            else:
                nums[target_index] = nums[start_index]
                number_to_search = nums[start_index]
                start_index = start_index + 1
                target_index = target_index + 1

        return target_index
        """
        if len(nums) < 2:
            return len(nums)

        curr = 0
        while curr + 1 < len(nums):
            if nums[curr] == nums[curr+1]:
                del nums[curr+1]
            else:
                curr += 1

        return len(nums)

if __name__ == '__main__':
    nums = [0,0,1,1,1,2,2,3,3,4]
    sol = Solution()
    res = sol.removeDuplicates(nums)
    print(res)

File: TopCoder/src/main/java/com/leetcode/RemoveDuplicates.java
package com.leetcode;

//https://leetcode.com/problems/remove-duplicates-from-sorted-array/
public class RemoveDuplicates {

    public int removeDuplicates(int[] nums) {
        if (nums == null ||nums.length == 0) {
            return 0;
        }
        int startIndex = 1;
        int targetIndex = 1;

        int numberToSearch = nums[0];

        while (startIndex < nums.length) {
            if (nums[startIndex] == numberToSearch) {
                startIndex++;
                continue;
            } else {
                nums[targetIndex] = nums[startIndex];
                targetIndex++;
                numberToSearch = nums[startIndex];
                startIndex++;
            }
        }
        return targetIndex ;
    }

    public static void main(String[] args) {
        int[] nums = {0,0,1,1,1,2,2,3,3,4};
        int res = new RemoveDuplicates().removeDuplicates(nums);
        System.out.println(res);
    }
}
